app/mailers/reminder_deadline_mailer.rb

```diff
@@ -3,7 +3,30 @@ class ReminderDeadlineMailer < ApplicationMailer
   def notify_deadline(partner)
     @partner = partner
     @organization = partner.organization
-    @deadline_day = partner.partner_group&.deadline_day_of_month || @organization.deadline_day
+    @deadline = deadline_date(partner)
+
     mail(to: @partner.email, subject: "#{@organization.name} Deadline Reminder")
   end
+
+  private
+
+  def deadline_date(partner)
+    today = Time.current.day
+
+    if partner.partner_group&.reminder_day_of_month == today && partner.partner_group&.deadline_day_of_month.present?
+      reminder_day = partner.partner_group.reminder_day_of_month
+      deadline_day = partner.partner_group.deadline_day_of_month
+    elsif partner.organization.reminder_day == today && partner.organization.deadline_day.present?
+      reminder_day = partner.organization.reminder_day
+      deadline_day = partner.organization.deadline_day
+    else
```

Comment:
Do you think this makes sense to encapsulate inside of the model?

Reply:
I definitely think so! Less we can put in the controller the better. Honestly I think this deserves to be in a service class since it spans multiple models (partners, partner groups and organizations).

Reply:
Agreed, this should be elsewhere.

Reply:
Wrote a service now.